src/unix/signal.c

```diff
@@ -53,8 +53,13 @@ RB_GENERATE_STATIC(uv__signal_tree_s,
                    uv_signal_s, tree_entry,
                    uv__signal_compare)
 
+static void uv__signal_global_reinit(void);
 
 static void uv__signal_global_init(void) {
+  if (!uv__signal_lock_pipefd[0])
+    if (pthread_atfork(NULL, NULL, &uv__signal_global_reinit))
+      abort();
```

Comment:
I'm not sure I understand this: is the reason for `if (!uv__signal_lock_pipefd[0])` that the `pthread_atfork()` needs only to be run on the parent process? Could you elaborate why? Thanks!